### PR TITLE
Add find task logic and test

### PR DESCRIPTION
**Pull request by jarrellc1**
Let's add findtask command to allow user to find task of a person.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Test
+    public void parse_validArgs_returnsFindTaskCommand() {
+        // Test with valid index argument
+        FindTaskCommand expectedFindTaskCommand = new FindTaskCommand(INDEX_FIRST_PERSON);
+        assertParseSuccess(parser, "1", expectedFindTaskCommand);
+
+        // Test with leading/trailing spaces
+        assertParseSuccess(parser, " 1 ", expectedFindTaskCommand);
+    }
```
Comment (yixianggg) [inline src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java:31]: Would it be better to test once with leading spaces and once with trailing spaces instead? So it is more specified.

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        // Test with non-numeric argument
+        assertParseFailure(parser, "abc", String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindTaskCommand.MESSAGE_USAGE));
+
+        // Test with negative number
+        assertParseFailure(parser, "-1", String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindTaskCommand.MESSAGE_USAGE));
+    }
+}
```
Comment (yixianggg) [inline src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java:41]: Good job covering these test cases!

File: src/main/java/seedu/address/logic/Messages.java [hunk 3]
```diff
@@ -21,6 +21,10 @@ public class Messages {
     public static final String MESSAGE_NONEXISTENT_PERSON =
             "The person '%1$s' does not exist in the system!";
     public static final String MESSAGE_INVALID_TASK_DISPLAYED_INDEX = "The task index provided is invalid";
+    public static final String MESSAGE_TASKS_LISTED_OVERVIEW = "%1$d tasks listed!";
+    public static final String MESSAGE_NO_TASKS_FOR_PERSON = "No tasks found for the person at index %1$d";
```
Comment (yixianggg) [inline src/main/java/seedu/address/logic/Messages.java:25]: Is this constant is used anywhere in the code?

File: src/main/java/seedu/address/logic/commands/FindTaskCommand.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        Person personToFindTasksFor = lastShownPersonList.get(targetIndex.getZeroBased());
+
+        // Filter tasks that are related to the selected person
+        model.updateFilteredTaskList(task -> task.getPatient().equals(personToFindTasksFor));
+
+        return new CommandResult(
+                String.format(Messages.MESSAGE_TASKS_LISTED_OVERVIEW, model.getFilteredTaskList().size()));
+    }
```
Comment (yixianggg) [inline src/main/java/seedu/address/logic/commands/FindTaskCommand.java:52]: Should there be a check that if getFilteredTaskList().size is 0, to display MESSAGE_NO_TASKS_FOR_PERSON?

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 5]
```diff
@@ -145,4 +146,29 @@ public void parseCommand_priorityMissingLevel_throwsParseException() {
...
+                        -> parser.parseCommand(FindTaskCommand.COMMAND_WORD + " abc"));
+    }
+
+    @Test
+    public void parseCommandFindTaskMissingIndexThrowsParseException() {
+        assertThrows(ParseException.class,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindTaskCommand.MESSAGE_USAGE), ()
+                        -> parser.parseCommand(FindTaskCommand.COMMAND_WORD + " "));
+    }
+
```
Comment (yixianggg) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:173]: Can consider a test case that appends some string value after a valid command? i.e. findtask 1 test

File: src/main/java/seedu/address/logic/Messages.java [hunk 6]
```diff
@@ -21,6 +21,10 @@ public class Messages {
     public static final String MESSAGE_NONEXISTENT_PERSON =
             "The person '%1$s' does not exist in the system!";
```
Comment (gavinsoh) [inline src/main/java/seedu/address/logic/Messages.java:22]: I like how you formatted the strings for error messages. This makes it easier to insert variable data into the messages, improving the clarity of error communication.

File: src/main/java/seedu/address/logic/commands/FindTaskCommand.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import seedu.address.model.person.Person;
+
+/**
+ * Finds and lists all tasks in the task list that are related to a specified person.
+ */
+public class FindTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "findtask";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all tasks related to the person identified by "
```
Comment (gavinsoh) [inline src/main/java/seedu/address/logic/commands/FindTaskCommand.java:20]: I like how you structured the detailed command usage. This is very efficient in  providing clear instructions on how to use the command, which greatly aids in user understanding and reduces user errors.

File: src/main/java/seedu/address/logic/commands/FindTaskCommand.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownPersonList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownPersonList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
```
Comment (gavinsoh) [inline src/main/java/seedu/address/logic/commands/FindTaskCommand.java:43]: Would it be better to add more context to the error message to help developers and users understand the issue better?

File: src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+        // different person -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_invalidPersonIndex_throwsCommandException() {
+        Index outOfBoundsIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        FindTaskCommand findTaskCommand = new FindTaskCommand(outOfBoundsIndex);
+
```
Comment (gavinsoh) [inline src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java:59]: I like how you implemented boundary testing by checking the behavior when an index is out of bounds. It ensures that the application robustly handles user errors, preventing runtime exceptions.

File: src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @Test
+    public void execute_validPersonIndex_noTasksFound() {
+        Index targetIndex = INDEX_SECOND_PERSON;
+        String expectedMessage = String.format(MESSAGE_TASKS_LISTED_OVERVIEW, 0);
+        FindTaskCommand command = new FindTaskCommand(targetIndex);
+        expectedModel.updateFilteredTaskList(task -> task.getPatient().equals(BOB));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredTaskList());
```
Comment (gavinsoh) [inline src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java:70]: I like how you have clearly laid out the expectations and verified that the model's state is correctly updated.

File: src/main/java/seedu/address/logic/commands/FindTaskCommand.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof FindTaskCommand)) {
+            return false;
+        }
+
+        FindTaskCommand otherFindTaskCommand = (FindTaskCommand) other;
+        return targetIndex.equals(otherFindTaskCommand.targetIndex);
```
Comment (xinweichong) [inline src/main/java/seedu/address/logic/commands/FindTaskCommand.java:66]: Will there be an issue with the `equals` method if the indexes of the individual tasks are updated as there are deletions to the tasklist?

File: src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java (hunk 1 above)
Comment (jarrellc1) [reply src/test/java/seedu/address/logic/parser/FindTaskCommandParserTest.java:31]: Thank you for the feedback, i will work on it.

File: src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        assertEquals(model.getFilteredTaskList(), expectedModel.getFilteredTaskList());
+    }
+
+    @Test
+    public void toStringMethod() {
+        FindTaskCommand findTaskCommand = new FindTaskCommand(INDEX_FIRST_PERSON);
+        String expected = "FindTaskCommand{targetIndex=" + INDEX_FIRST_PERSON + "}";
+        assertEquals(expected, findTaskCommand.toString());
+    }
+}
```
Comment (benjiBase) [inline src/test/java/seedu/address/logic/commands/FindTaskCommandTest.java:90]: I like that the test coverage handles different situations that can during the runtime of app

**Pull request merged**